### PR TITLE
[FVM]SetProgram can be called with nil

### PR DESCRIPTION
**Pull request by janezpodhostnik**
relates to: https://github.com/onflow/flow-go/issues/3802
uses cadence from: https://github.com/onflow/cadence/pull/2241

If `GetProgram` is called and returns nil `SetProgram` must be called. Cadence recently changed that some parsing errors that occur during program loading are not critical (and cadence execution continues). In this case `SetProgram` mist be called with a `nil` to signal that program loading is done, but no program was loaded. 

Should not be merged to master. Master should get the proper fix of changing `get` and `set` into a `getOrSet(loadProgramClosure)`

### CODE REVIEW

**Review by pattyshack — APPROVED** [review]

**Review by ramtinms — APPROVED** [review]

**Comment by turbolent** [comment]
Nice! Could you please also update to https://github.com/onflow/flow-go-sdk/releases/tag/v0.31.3 ?

File: fvm/fvm_test.go [hunk 1]
```diff
@@ -2114,3 +2114,120 @@ func TestInteractionLimit(t *testing.T) {
...
+				})
+				return paths
+			}`,
+			)).WithArguments(
+				jsoncdc.MustEncode(address),
+			)
+			scriptCtx := fvm.NewContextFromParent(ctx)
+			scriptCtx.DerivedBlockData = nil
+			err = vm.Run(scriptCtx, script, view)
+
```
Comment (turbolent) [inline fvm/fvm_test.go:2230]: Maybe also add an assertion about the value returned from the script

**Review by turbolent — APPROVED** [review]

**Comment by turbolent** [comment]
@peterargue Do we also need this PR against `v0.29-no-pacemaker`?

**Comment by peterargue** [comment]
> @peterargue Do we also need this PR against `v0.29-no-pacemaker`?

yes, please

**Comment by turbolent** [comment]
#3812

**Comment by turbolent** [comment]
bors merge

**Pull request merged**